drawablepainter/src/main/java/com/google/accompanist/drawablepainter/DrawablePainter.kt

```diff
@@ -121,15 +129,10 @@ class DrawablePainter(
     }
 
     override val intrinsicSize: Size
-        get() = when {
-            // Only return a finite size if the drawable has an intrinsic size
-            drawable.intrinsicWidth >= 0 && drawable.intrinsicHeight >= 0 -> {
-                Size(
-                    width = drawable.intrinsicWidth.toFloat(),
-                    height = drawable.intrinsicHeight.toFloat(),
-                )
-            }
-            else -> Size.Unspecified
+        get() {
+            // Reading this ensures that we re-layout when invalidateDrawable() is called
+            layoutTick
+            return drawable.intrinsicSize.also { lastIntrinsicSize = it }
```

Comment:
it feels like a source of truth issue to me. seems like everytime the size of the drawable changes we get the callback. which means we can have our own source of truth here

```
override var intrinsicSize: Size by mutableStateOf(drawable.intrinsicSize)
     private set
...
            override fun invalidateDrawable(d: Drawable) {
                intrinsicSize = drawable.intrinsicSize
            }
```

this feels way cleaner for me compared to the tick approach. wdyt?

Reply:
> Ready to merge!

Sorry, but I left one more comment 2 days ago, you probably didn't notice it

Reply:
Woops missed this.

> seems like everytime the size of the drawable changes we get the callback

Not quite. `invalidateDrawable()` will only be called when the drawable itself requests that it needs to be invalidated. It won't be called if Compose triggers the draw pass.

The current code is the right place, as it will update the size on any Compose draw pass. Moving it to `invalidateDrawable` will mean that it only updates the size if the drawable requests it (which may never happen) for static drawables.

Reply:
But don't you agree that if such callback is not applied everything will work unreliable. Like, ok, the size of the drawable is changing and we don't get invalidateDrawable callback. And if(and when) someone will call DrawablePainter.intrinsicSize we will return the new size. But if nothing will trigger that we do not update the ui.
This makes me think that we should have callback to be always working. I see that in current impl we set callback in onRemembered and reset in onForgotten. So if the painter is not remembered nothing works. Maybe it is not what we want? I see that Drawable.setCallback is actually wrapping the passed object into a weak reference internally. so maybe we will instead set the callback right in the init block of DrawablePainter, so we are guaranteed to receive this callback when the size changes. And then refactor to have a state of size instead of a ticker